src/classes/BookingHandler2.cls

```diff
@@ -0,0 +1,777 @@
+public class BookingHandler2 {
```

Comment:
This class seems to have grown very large and is becoming difficult to maintain. I suggest we consider refactoring to reduce the number of methods.